File: Kotlin/3_Longest_Substring_Without_Repeating_Characters.kt
class Solution {
    fun lengthOfLongestSubstring(s: String): Int {
        var result = 0
        var start = hashMapOf<Char, Int>()
        var j = 0
        for(i in s.indices) {
            if(start.containsKey(s[i])){
                j = Math.max(j, start[s[i]]!! + 1)
            }
            start[s[i]] = i
            result = Math.max(result, i - j + 1)
        }
        return result
    }
}

File: Cpp/3_Longest_Substring_Without_Repeating_Characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        map<char, int> start;
        int result = 0;
        int j = 0;
        for(int i=0; i<s.length(); ++i){
            auto find_char = start.find(s.at(i));
            if (find_char != start.end()){
                j = std::max(j, find_char->second + 1);

            }
            start[s.at(i)] = i;
            result = std::max(result, i - j + 1);
        }
        
        return result;
    }
};